controllers/bootstrap/init.go

```diff
@@ -196,11 +196,9 @@ func (b *Bootstrap) InitResources(instance *apiv3.CommonService) error {
 	// Check storageClass
 	csStorageClass, err := util.GetStorageClass(b.Reader)
 	if err != nil {
-		return fmt.Errorf("failed to get StorageClass")
-	}
-	if err := util.ValidateStorageClass(csStorageClass); err != nil {
-		return err
+		klog.Info("failed to get StorageClass")
```

Comment:
I think we should keep `return fmt.Errorf("failed to get StorageClass")` here. I think it is the error that we failed to List storageclass from the cluster, not because there is no Storageclass avaliable.

Also because we will only inform the user, not stop the reconcile.

I think we need to add a separate the error check here
one is `errors.IsNotFound(err)` and the other is the remaining errors
https://github.com/IBM/ibm-common-service-operator/blob/master/controllers/common/util.go#L225

/cc @horis233 Please correct me if it is not correct.

Reply:
I see. So `return fmt.Errorf("failed to get StorageClass")` this error indicates `not found StorageClass attribute` right? not about StorageClass instance.

Reply:
yeah, kind of.

The error may happen when there is issue to communicate with kube apiserver. it could be permission error, network err or something else. It will be better to return that error as well.
`return fmt.Errorf("failed to get StorageClass: %v", err)`

Reply:
make sense, if those kinds of error happens, do we want to stop the reconcile or continue? I think if we `return fmt.Errorf("failed to get StorageClass: %v", err)` in controllers/bootstrap/init.go, since **return**, it will stop the reconcile.

Reply:
No, we wanna continue the reconciliation.

If the reconciler receive the return error, it will requeue the request automatically.

Reply:
Thanks for the explanation! Updated with we discussed above, please review.